internal/pkg/externalplugins/cherrypicker/cherrypicker.go

```diff
@@ -0,0 +1,685 @@
+			if handledBranches[targetBranch] {
+				// Branch already handled. Skip.
+				continue
+			}
+			if targetBranch == baseBranch {
+				resp := fmt.Sprintf("base branch (%s) needs to differ from target branch (%s).", baseBranch, targetBranch)
+				l.Info(resp)
+				if err := s.createComment(l, org, repo, num, ic, resp); err != nil {
+					l.WithError(err).WithField("response", resp).Error("Failed to create comment.")
+				}
+				continue
+			}
+			handledBranches[targetBranch] = true
+			l := l.WithFields(logrus.Fields{
+				"requestor":     requestor,
+				"target_branch": targetBranch,
+			})
+			l.Debug("Cherrypick request.")
+			err := s.handle(l, requestor, ic, org, repo, targetBranch, &pr)
+			if err != nil {
+				l.WithError(err).Error("failed to create cherrypick")
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (s *Server) handle(logger *logrus.Entry, requestor string,
+	comment *github.IssueComment, org, repo, targetBranch string, pr *github.PullRequest) error {
+	num := pr.Number
+	title := pr.Title
+	body := pr.Body
+	var lock *sync.Mutex
+	func() {
+		s.mapLock.Lock()
+		defer s.mapLock.Unlock()
+		if _, ok := s.lockMap[cherryPickRequest{org, repo, num, targetBranch}]; !ok {
+			if s.lockMap == nil {
+				s.lockMap = map[cherryPickRequest]*sync.Mutex{}
+			}
+			s.lockMap[cherryPickRequest{org, repo, num, targetBranch}] = &sync.Mutex{}
+		}
+		lock = s.lockMap[cherryPickRequest{org, repo, num, targetBranch}]
+	}()
+	lock.Lock()
+	defer lock.Unlock()
+
+	opts := s.ConfigAgent.Config().CherrypickerFor(org, repo)
+
+	forkName, err := s.ensureForkExists(org, repo)
+	if err != nil {
+		logger.WithError(err).Warn("failed to ensure fork exists")
+		resp := fmt.Sprintf("cannot fork %s/%s: %v.", org, repo, err)
+		return s.createComment(logger, org, repo, num, comment, resp)
+	}
+
+	// Clone the repo, checkout the target branch.
+	startClone := time.Now()
+	r, err := s.GitClient.ClientFor(org, repo)
+	if err != nil {
+		return fmt.Errorf("failed to get git client for %s/%s: %w", org, forkName, err)
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			logger.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.Checkout(targetBranch); err != nil {
+		logger.WithError(err).Warn("failed to checkout target branch")
+		resp := fmt.Sprintf("cannot checkout `%s`: %v.", targetBranch, err)
+		return s.createComment(logger, org, repo, num, comment, resp)
+	}
+	logger.WithField("duration", time.Since(startClone)).Info("Cloned and checked out target branch.")
+
+	// Fetch the patch from GitHub
+	localPath, err := s.getPatch(org, repo, targetBranch, num)
+	if err != nil {
+		return fmt.Errorf("failed to get patch: %w", err)
+	}
+
+	// Setup git name and email.
+	if err := r.Config("user.name", s.BotUser.Login); err != nil {
+		return fmt.Errorf("failed to configure git user: %w", err)
+	}
+	email := s.Email
+	if email == "" {
+		email = s.BotUser.Email
+	}
+	if err := r.Config("user.Email", email); err != nil {
```

Comment:
Maybe `Email` should use lowercase uniformly.

Reply:
Fixed.